### PR TITLE
BUGFIX: rejectif.NsAtApex never matched the apex label

### PR DESCRIPTION
**Pull request by SimenBai**
## Problem

`rejectif.NsAtApex` is a no-op. It checks `rc.GetLabel() == ""`, but the apex label is `"@"` (per `models/record.go`: *"the apex is returned as @"*), so it never returns an error. Every other apex check in the tree uses `"@"` — including the sibling helpers in this same package (`label.go`, `dnskey.go`) and dozens of providers.

The three providers that use `NsAtApex` — DOMAINNAMESHOP, JOKER, VERCEL — all rely on it to reject NS records at the apex. The bug was masked for JOKER and VERCEL because they're *also* excluded from the apex-NS integration test via `not()`. DOMAINNAMESHOP is not, so `NS_only_APEX:Single_NS_at_apex` fails with `400 - DNS record failed validation` instead of being skipped.

This is the follow-up to #4702, which added `a.Add("NS", rejectif.NsAtApex)` to DOMAINNAMESHOP but had no effect because of this latent helper bug.

## Fix

Match `"@"` (and `""` defensively). Add a unit test — the helper had none, which is why the bug went unnoticed.

## Verification

```
go test ./pkg/rejectif/ -run TestNsAtApex -v
```

All four cases pass (`@` and empty → rejected; subdomains → allowed). With this change, the DOMAINNAMESHOP apex-NS integration test now **skips** instead of failing, and subdomain NS is unaffected.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01VESo8kcAdjSvRxNWetq8Ru

### CODE REVIEW

File: pkg/rejectif/ns.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ import (
 // NsAtApex detects NS records at the apex/root domain.
 // Use this when a provider doesn't support custom NS records at the apex.
 func NsAtApex(rc *models.RecordConfig) error {
-	if rc.GetLabel() == "" {
+	// The apex label is normally "@", but accept "" defensively too.
+	if rc.GetLabel() == "@" || rc.GetLabel() == "" {
```
Comment (TomOnTime) [inline pkg/rejectif/ns.go]: GetLabel() should never return "".  If it does, we need to track down how that happened and fix it.  Do you have a case where it happened or is this just speculative?

Comment (SimenBai) [reply pkg/rejectif/ns.go]: No observed case; it's not from a bug I can reproduce. I kept `== ""` only to preserve the original check's behavior rather than change it. Updating the code to reflect that only @ is expected
